drivers/storage/portworx/component/csi.go

```diff
@@ -501,11 +504,13 @@ func (c *csi) createDeployment(
 		)
 	}
 
-	updatedTopologySpreadConstraints, err := util.GetTopologySpreadConstraints(c.k8sClient, csiDeploymentTemplateLabels)
+	updatedTopologySpreadConstraints, err := util.GetTopologySpreadConstraints(c.k8sClient, csiDeploymentTemplateSelectorLabels)
 	if err != nil {
 		return err
 	}
 
+	deployment := getCSIDeploymentSpec(cluster, csiConfig, ownerRef, provisionerImage, attacherImage,
```

Comment:
Any reason this was moved out of the original `if` block?

Reply:
Ah, this was from when I was checking for diffs and updating existing deployments. I will revert changes like this.